daft/execution/execution_step.py

```diff
@@ -947,6 +947,40 @@ def run_partial_metadata(self, input_metadatas: list[PartialPartitionMetadata])
         ]
 
 
+@dataclass(frozen=True)
```

Comment:
is this frozen so that it can be sent over I/O / serialized?

Reply:
not sure, but there's no need to ever change it. just something I noticed other ops also have so I added it too